cmd/admin-handlers.go

```diff
@@ -534,6 +612,44 @@ func writeSetConfigResponse(w http.ResponseWriter, peers adminPeers,
 	return
 }
 
+func (a adminAPIHandlers) setConfigDotJSON(ctx context.Context, configBytes []byte) ([]error, error) {
+	// Write config received from request onto a temporary file on
+	// all nodes.
+	tmpFileName := fmt.Sprintf(minioConfigTmpFormat, mustGetUUID())
+	errs := writeTmpConfigPeers(globalAdminPeers, tmpFileName, configBytes)
+
+	// Check if the operation succeeded in quorum or more nodes.
+	rErr := reduceWriteQuorumErrs(ctx, errs, nil, len(globalAdminPeers)/2+1)
+	if rErr != nil {
+		return errs, rErr
+	}
+
+	// Take a lock on minio/config.json. NB minio is a reserved
+	// bucket name and wouldn't conflict with normal object
+	// operations.
+	configLock := globalNSMutex.NewNSLock(minioReservedBucket, minioConfigFile)
+	if configLock.GetLock(globalObjectTimeout) != nil {
+		return nil, errOperationTimedOut
+	}
+	defer configLock.Unlock()
+
+	// Save a backup of the current config
+	errs = backupConfigPeers(globalAdminPeers)
+	rErr = reduceWriteQuorumErrs(ctx, errs, nil, len(globalAdminPeers)/2+1)
+	if rErr != nil {
+		return errs, rErr
+	}
+
+	// Rename the temporary config file to config.json
+	errs = commitConfigPeers(globalAdminPeers, tmpFileName)
```

Comment:
Once there are errors while writing new `config.json` to a temp location, we should not try to move that temp `config.json` on the nodes where it failed.

Reply:
> Once there are errors while writing new config.json to a temp location, we should not try to move that temp config.json on the nodes where it failed.

I didn't understand exactly your point here

Reply:
We should not be trying to write on nodes where `writeTmpConfigPeers` failed. You will have unintended consequence